### PR TITLE
Update copyright year

### PR DESCRIPTION
**Pull request by jbakosi**
Large but mostly trivial change-set. Note that not only the year got updated (2017->2018) but also modernized the update script a little.

### CODE REVIEW

File: script/update_copyright.sh [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+#   quinoa-cmake: .
 #
 ################################################################################
 
-find <dirs> -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | xargs sed -i 's/opyright 2012-2015, J. Bakosi, 2016-2017/opyright 2012-2015, J. Bakosi, 2016-2018/g'
-find <dirs> -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | xargs sed -i 's/Copyright (c) 2016-2017/Copyright (c) 2016-2018/g'
-find <dirs> -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | xargs sed -i 's/Copyright 2016-2017/Copyright 2016-2018/g'
+# Require at least one argument
+die () { echo >&2 "$@"; exit 1; }
+[ "$#" -eq 0 ] && die "Usage: $0 <directories>"
```
[the remark this answers is not in the document]
Comment (jbakosi) [reply script/update_copyright.sh]: Whoa! An "empty" as in a space? Ah that is twisted. ;-) Okay, will try it.

File: script/update_copyright.sh [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+
+declare -a dirs=( "$@" )
+
+for i in "${dirs[@]}"
```
[the remark this answers is not in the document]
Comment (jbakosi) [reply script/update_copyright.sh]: Oh, nice! Will try that too.

File: script/update_copyright.sh [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+do
+  find $i -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | grep -v update_copyright.sh | xargs sed -i 's/opyright 2012-2015, J. Bakosi, 2016-2017/opyright 2012-2015, J. Bakosi, 2016-2018/g'
```
[the remark this answers is not in the document]
Comment (jbakosi) [reply script/update_copyright.sh]: Nice!

File: script/update_copyright.sh [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
+#   quinoa-cmake: .
 #
 ################################################################################
 
-find <dirs> -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | xargs sed -i 's/opyright 2012-2015, J. Bakosi, 2016-2017/opyright 2012-2015, J. Bakosi, 2016-2018/g'
-find <dirs> -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | xargs sed -i 's/Copyright (c) 2016-2017/Copyright (c) 2016-2018/g'
-find <dirs> -type f -exec grep -Iq . {} \; -and -print | cut -f1 -d: | xargs sed -i 's/Copyright 2016-2017/Copyright 2016-2018/g'
+# Require at least one argument
+die () { echo >&2 "$@"; exit 1; }
+[ -z $1 ] && die "Usage: $0 <directories>"
```
Comment (junghans) [inline script/update_copyright.sh]: You should add a `#! /bin/bash` to the top and use `[[ .. ]]` here.

**Pull request merged**